src/core/evaluation/expressionHandler.ts

```diff
@@ -69,25 +76,54 @@ export default class ExpressionHandler {
     return "";
   }
 
-  private resolveSymbol(name: string): string {
-    // Symbols should always be cell references in our case.
+  private resolveSymbol(symbol: string): any {
     // TODO only handling references within one sheet for now.
-    const split = name.split(/[0-9]+/);
-    if (split.length != 2) throw new Error("Invalid symbol: " + name);
-
-    // Resolve column/row indices from text-based cell reference.
-    const columnStr = split[0];
-    const rowStr = name.substring(columnStr.length);
-    const columnIndex = ExpressionHandler.resolveColumnIndex(columnStr);
-    if (columnIndex == -1) throw new Error("Invalid symbol: " + name);
-    const rowIndex = parseInt(rowStr) - 1;
+    if (symbol.includes(":")) {
+      const rangeParts = symbol.split(":").filter((s) => s !== "");
+      if (rangeParts.length != 2) throw new Error("Invalid range: " + symbol);
+
+      const start = ExpressionHandler.parseSymbol(rangeParts[0]);
+      const end = ExpressionHandler.parseSymbol(rangeParts[1]);
+
+      const values: string[] = [];
+      for (let i = start.rowIndex; i <= end.rowIndex; i++) {
+        for (let j = start.colIndex; j <= end.colIndex; j++) {
+          const cellInfo = this.sheet.getCellInfoAt(j, i);
+          if (cellInfo == null || cellInfo.state != CellState.OK)
+            throw new Error("Invalid cell reference: " + symbol);
+
+          this.cellRefHolder.push(cellInfo.position);
+          values.push(cellInfo.value!);
+        }
+      }
+      return values;
+    } else {
+      const { colIndex, rowIndex } = ExpressionHandler.parseSymbol(symbol);
+
+      const cellInfo = this.sheet.getCellInfoAt(colIndex, rowIndex);
+      if (cellInfo == null || cellInfo.state != CellState.OK)
+        throw new Error("Invalid cell reference: " + symbol);
+
+      this.cellRefHolder.push(cellInfo.position);
+      return cellInfo.value!;
+    }
+  }
```

Comment:
Could remove the else statement (nesting) since the above if-block always returns.